tests/behaviour/qbwriter.feature

```diff
@@ -404,12 +449,55 @@ Feature: Test outputting CSV-Ws with Qb flavouring.
       """
     And the RDF should pass "skos, qb" SPARQL tests
 
-  Scenario: A cube with an option attribute which has missing data values should validate successfully
-    Given a single-measure QbCube named "Some Qube" with optional attribute values missing
+  Scenario: A cube with an optional attribute which has missing data values should validate successfully given ATTRIBUTE_VALUE_CODELISTS is False
+    Given the ATTRIBUTE_VALUE_CODELISTS feature flag is set to False
+    And a single-measure QbCube named "Some Qube" with optional attribute values missing
     Then the CSVqb should pass all validations
     When the cube is serialised to CSV-W
     Then csvwcheck validation of all CSV-Ws should succeed
 
+  Scenario: Observation Values are Optional where an 'sdmxa:ObsStatus' Attribute is Present given ATTRIBUTE_VALUE_CODELISTS is False
+    Given the ATTRIBUTE_VALUE_CODELISTS feature flag is set to False
+    And a single-measure QbCube named "Good Qube" with missing observation values and `sdmxa:obsStatus` replacements
+    Then the CSVqb should pass all validations
+    When the cube is serialised to CSV-W
+    Then csvlint validation of "good-qube.csv-metadata.json" should succeed
+
+  Scenario: Observation Values are Required where an 'sdmxa:ObsStatus' Attribute Column is present but no value is set given ATTRIBUTE_VALUE_CODELISTS is False.
+    Given the ATTRIBUTE_VALUE_CODELISTS feature flag is set to False
+    And a single-measure QbCube named "Bad Qube" with missing observation values and missing `sdmxa:obsStatus` replacements
+    Then the CSVqb should fail validation with "Missing value(s) found for 'Value' in row(s) 0"
+    When the cube is serialised to CSV-W
+    # Unfortunately, CSV-W validation will *not* catch this error since the obs column cannot be marked as `required`
+    # since an `sdmxa:obsStatus` Attribute column has been defined.
+    Then csvlint validation of "bad-qube.csv-metadata.json" should succeed
+
+# TODO 820 Fix this test
+  # Scenario: A cube with an optional attribute which has missing data values should validate successfully given ATTRIBUTE_VALUE_CODELISTS is True
+  #   Given the ATTRIBUTE_VALUE_CODELISTS feature flag is set to True
+  #   And a single-measure QbCube named "Some Qube" with optional attribute values missing
+  #   Then the CSVqb should pass all validations
+  #   When the cube is serialised to CSV-W
+  #   Then csvlint validation of all CSV-Ws should succeed
+
+# TODO 820 Fix this test
```

Comment:
Follow up this ticket with a new one, which makes sure we support a sparse attribute column (i.e. obsStatus) when we have an attribute column with a code list (and thus is dense).

(This test can obvs go.)

Reply:
Created new ticket (#873) and added to epic